### PR TITLE
feat(autoware_ad_api_specs): define routing interface

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

The interface definition for https://github.com/autowarefoundation/autoware.universe/pull/1494, https://github.com/autowarefoundation/autoware.universe/pull/1495.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by kenji-miyake** [comment]
Waiting for comments in https://github.com/autowarefoundation/autoware.universe/pull/1494 to be resolved and reflected in this PR.

**Comment by isamu-takagi** [comment]
All comments in  https://github.com/autowarefoundation/autoware.universe/pull/1494 are resolved.

File: common/component_interface_specs/include/component_interface_specs/planning.hpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+struct ClearRoute
+{
+  using Service = autoware_ad_api_msgs::srv::ClearRoute;
+  static constexpr char name[] = "/planning/mission_planning/clear_route";
+};
+
+struct RouteState
+{
+  using Message = autoware_ad_api_msgs::msg::RouteState;
+  static constexpr char name[] = "/planning/mission_planning/state";
```
Comment (kenji-miyake) [inline common/component_interface_specs/include/component_interface_specs/planning.hpp:50]: I still feel `route_state` is clearer and better, but I'll leave it to you.
https://github.com/autowarefoundation/autoware.universe/pull/1494#discussion_r940924287

**Review by kenji-miyake — APPROVED** [review]

File: common/component_interface_specs/include/component_interface_specs/planning.hpp (hunk 1 above)
Comment (kenji-miyake) [reply common/component_interface_specs/include/component_interface_specs/planning.hpp:50]: @yukkysaito @mitsudome-r Please comment if you have any thoughts.

Comment (isamu-takagi) [reply common/component_interface_specs/include/component_interface_specs/planning.hpp]: Done https://github.com/autowarefoundation/autoware.universe/pull/1559/commits/cc37ba6ae986c75f2ae8d938742b53dba79850a3.

**Review by kenji-miyake — APPROVED** [review]

**Pull request merged**